tests/python/gaia-ui-tests/gaiatest/apps/contacts/app.py

```diff
@@ -157,5 +157,5 @@ def _return_class_from_tap(self, return_class='ContactDetails'):
                 return None
             else:
                 # We are using contacts picker in activity - after choosing, fall back to open app
-                Wait(self.marionette).until(lambda m: self.apps.displayed_app.name != Contacts.name)
+                Wait(self.marionette).until(lambda m: self.apps.displayed_app.manifest_url != Contacts.manifest_url)
```

Comment:
We can factorize this method too, in `Base`, exactly the same way as we did in `wait_to_be_displayed()`